packages/flutter/lib/src/material/action_buttons.dart

```diff
@@ -0,0 +1,427 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'action_icons_theme.dart';
+import 'button_style.dart';
+import 'debug.dart';
+import 'icon_button.dart';
+import 'icons.dart';
+import 'material_localizations.dart';
+import 'scaffold.dart';
+import 'theme.dart';
+
+abstract class _ActionButton extends StatelessWidget {
+  /// Creates a Material Design icon button.
+  const _ActionButton({
+    super.key,
+    this.color,
+    required this.icon,
+    this.iconSize,
+    required this.onPressed,
+    this.style,
+  });
+
+  /// The icon to display inside the button.
+  final Widget icon;
+
+  /// The callback that is called when the button is tapped
+  /// or otherwise activated.
+  ///
+  /// If this is set to null, the button will do a default action
+  /// when it is tapped or activated.
+  final VoidCallback? onPressed;
+
+  /// The color to use for the icon.
+  ///
+  /// Defaults to the [IconThemeData.color] specified in the ambient [IconTheme],
+  /// which usually matches the ambient [Theme]'s [ThemeData.iconTheme].
+  final Color? color;
+
+  /// The size to use for the icon.
+  ///
+  /// Defaults to the [IconThemeData.size] specified in the ambient [IconTheme],
+  /// which usually matches the ambient [Theme]'s [ThemeData.iconTheme].
+  final double? iconSize;
```

Comment:
This should be a private property. It's not a constructor parameter in the public classes, but developers can use the style parameter to specify [ButtonStyle.iconSize](https://api.flutter.dev/flutter/material/ButtonStyle/iconSize.html)